File: python/src/python_robotics_algorithms/pointcloud/transform.py
import numpy as np
from scipy.spatial.transform import RigidTransform as Tf
from scipy.spatial.transform import Rotation as R
import timeit


def SetUp(numpoints):
    points = np.random.rand(numpoints, 3).astype(np.float32)
    t = np.random.rand(3).astype(np.float32)
    r = R.from_euler("XYZ", [90, 30, 0], degrees=True)
    transform = Tf.from_components(t, r)
    return points, transform

for num_points in [8000, 16000, 32000]:
    setup_code = f"points, transform = SetUp({num_points})" 
    run_code = "transform.apply(points)"

    print(f"Num points: {num_points}, time: {timeit.timeit(run_code, setup=setup_code, number=1000, globals=globals())} ms")

File: julia/pointcloud/transform.jl
using LinearAlgebra
using CoordinateTransformations
using Random
using BenchmarkTools

const SIZES = (8000, 16000, 32000)


R = Float32[1.0 0.0 0.0; 0.0 1.0 0.0; 0.0 0.0 1.0]
t = Float32[2.0, 2.0, 4.0]
A = AffineMap(R, t)

for s in SIZES
    points = []
    for _ in 1:s
        p = rand(Float32, 3)
        push!(points, p)
    end

    b = @benchmark [$A(p) for p in $points]

    println(b)

end